### DISCUSSION TITLE
Can you (from within PMF) evaluate a PMF function with arrayIndex set to the layer index of some depth?

### DISCUSSION THREAD

**Discussion by ilhuber** [APSIMInitiative/ApsimX#9473] [Q&A] [answered]
Is there some PMF way to get a functions value forcing the arrayIndex to be a certain value (say layer index of sowing depth)? My goal is to evaluate SoilWaterScale at the sowing depth layer as a way of applying some penalty to pre-emergence thermal time within the phenology of a plant module. The SoilWaterScale result will be the XValue of some LinearInterpolationFunction. Maybe there is a smarter way to handle this.

**Comment by hol353** [comment, marked as answer]
The SoilWaterScale function takes an arrayIndex value. You could call SoilWaterScale.Value from an manager script with the layer index of the seed depth. The manager script could then call MathUtilities.LinearInterpReal. I can't think of another more PMF way of doing what you want.
